### ISSUE TITLE
Wayland platform keymap glitch

### ISSUE THREAD

**Issue by wmww** [MirServer/mir#2438]
When I run Mir-on-Wayland on top of Mir-on-X and I open lxterminal, I can not type into the terminal that's in the nested Mir. If I keysmash sometimes garbage gets typed. This behavior is unaffected by #2437.

**Comment by AlanGriffiths** [comment]
I cannot reproduce:

```
export WAYLAND_DISPLAY=wayland-99
ubuntu-frame&
miral-app -terminal lxterminal --wayland-host $WAYLAND_DISPLAY
```

it all works fine.

So how are _you_ setting this up?

**Comment by wmww** [comment]
Hmmmm, thanks for testing.
```
WAYLAND_DISPLAY=outer ./bin/miral-shell&
WAYLAND_DISPLAY=inner DISPLAY=no ./bin/miral-shell --wayland-host outer&
WAYLAND_DISPLAY=inner gedit
```
Gedit works, however as soon as you hit Ctrl+Alt+T terminals appear in both outer and inner Mir's, and thereafter neither of the inner apps accept keyboard input, I will investigate further.

**Comment by AlanGriffiths** [comment]
OK, can reproduce.

I'm actually surprised that the outer shell both handles `Ctrl-T` _and_ passes it to the inner shell.

Interestingly, I can start `lxterminal` from the terminal without seeing issues:

```
WAYLAND_DISPLAY=inner lxterminal&
WAYLAND_DISPLAY=outer lxterminal&
```

**Comment by wmww** [comment]
It appears to be a modifier problem. wayland-debug shows that modifiers are not being automatically released, and pressing ctrl and alt again with the inner window focused seems to fix it. More investigation needed to determine exactly where it's going wrong.

**Comment by AlanGriffiths** [comment]
> More investigation needed to determine exactly where it's going wrong.

My guess would be that the "outer" shell is consuming part of the `Ctrl+Alt+T` event when it reacts to it but also (incorrectly IMO) passes through part of it. Then the inconsistent event stream leaves the "inner" shell input stack in confusion.

**Comment by AlanGriffiths** [comment]
Fixed by #2537
